src/common/mfu_daos.c

```diff
@@ -1881,3 +1912,2856 @@ int mfu_daos_flist_sync(
+    if (hdf5->single_dspace < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create single_dspace");
+        rc = 1;
+        goto out;
+    }
+    err = H5Pset_layout(hdf5->plist, H5D_CHUNKED);
+    if (err < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to set property layout");
+        rc = 1;
+        goto out;
+    }
+    err = H5Pset_chunk(hdf5->plist, 1, hdf5->rx_chunk_dims);
+    if (err < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to set chunk size");
+        rc = 1;
+        goto out;
+    }
+out:
+    if (rc != 0) {
+        H5Tclose(hdf5->usr_attr_memtype);
+        H5Tclose(hdf5->usr_attr_name_vtype);
+        H5Tclose(hdf5->usr_attr_val_vtype);
+        H5Tclose(hdf5->dkey_memtype);
+        H5Tclose(hdf5->dkey_vtype);
+        H5Tclose(hdf5->akey_memtype);
+        H5Tclose(hdf5->akey_vtype);
+    }
+    return rc;
+}
+
+static int serialize_akeys(struct hdf5_args *hdf5,
+                           daos_key_t diov,
+                           uint64_t *dk_index,
+                           uint64_t *ak_index,
+                           uint64_t *total_akeys,
+                           daos_handle_t *oh)
+{
+    int             rc = 0;
+    herr_t          err = 0;
+    int             j = 0;
+    daos_anchor_t   akey_anchor = {0}; 
+    d_sg_list_t     akey_sgl;
+    d_iov_t         akey_iov;
+    daos_key_desc_t akey_kds[ENUM_DESC_NR] = {0};
+    uint32_t        akey_number = ENUM_DESC_NR;
+    char            akey_enum_buf[ENUM_DESC_BUF] = {0};
+    char            akey[ENUM_KEY_BUF] = {0};
+    char            *akey_ptr = NULL;
+    daos_key_t      aiov;
+    daos_iod_t      iod;
+    char            rec_name[32];
+    int             path_len = 0;
+    int             size = 0;
+    hvl_t           *akey_val;
+
+    while (!daos_anchor_is_eof(&akey_anchor)) {
+        memset(akey_kds, 0, sizeof(akey_kds));
+        memset(akey, 0, sizeof(akey));
+        memset(akey_enum_buf, 0, sizeof(akey_enum_buf));
+        akey_number = ENUM_DESC_NR;
+
+        akey_sgl.sg_nr     = 1;
+        akey_sgl.sg_nr_out = 0;
+        akey_sgl.sg_iovs   = &akey_iov;
+
+        d_iov_set(&akey_iov, akey_enum_buf, ENUM_DESC_BUF);
+
+        /* get akeys */
+        rc = daos_obj_list_akey(*oh, DAOS_TX_NONE, &diov,
+                                &akey_number, akey_kds,
+                                &akey_sgl, &akey_anchor, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to list akeys: %d", rc);
+            goto out;
+        }
+
+        /* if no akeys returned move on */
+        if (akey_number == 0)
+            continue;
+
+        size = (akey_number + *total_akeys) * sizeof(akey_t);
+        *hdf5->ak = realloc(*hdf5->ak, size);
+        if (*hdf5->ak == NULL) {
+            rc = ENOMEM;
+            goto out;
+        }
+
+        /* parse out individual akeys based on key length and
+         * numver of dkeys returned
+         */
+        (*hdf5->dk)[*dk_index].akey_offset = *ak_index;
```

Comment:
@dsikich We need to set this BEFORE the while loop above. Otherwise, it gets overwritten